### PR TITLE
Docs: Grafana 8 alerting default 

### PR DESCRIPTION
**Pull request by yuri-tceretian**
**What this PR does / why we need it**:
This PR is to update documentation according to the changes in the configuration and code introduced in https://github.com/grafana/grafana/pull/41380.
Also, some of the changes are related to https://github.com/grafana/grafana/pull/40540 that introduced special alert instances for NoData

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by achatterjee-grafana** [comment]
Since Grafana 8 alerting will be the default in 8.3 GA, is it time to remove legacy dashboard alerting documentation? Thanks for any clarification @pkolyvas and alerting team!

File: docs/sources/alerting/unified-alerting/migration.md [hunk 1]
```diff
@@ -1,48 +1,29 @@
 +++
-title = "Opt-in to Grafana 8 alerting"
+title = "Migration to Grafana 8 alerting"
 description = "Enable Grafana 8 Alerts"
 weight = 115
 +++
 
-# Opt-in to Grafana 8 alerting
```
Comment (achatterjee-grafana) [inline docs/sources/alerting/unified-alerting/migration.md:7]: Can users opt in to use legacy alerting if they want to? Or they can no longer use legacy alerting?

File: docs/sources/alerting/old-alerting/view-alerts.md [hunk 2]
```diff
@@ -8,6 +8,8 @@ aliases = ["/docs/grafana/latest/alerting/view-alerts/"]
 
 # View existing alert rules
 
+>**Note**: Deprecated. Refer to [Grafana 8 alerting]({{< relref "../unified-alerting/_index.md" >}}) for documentation
```
Comment (achatterjee-grafana) [inline docs/sources/alerting/old-alerting/view-alerts.md:11]: Should we remove legacy alerting docs?

File: docs/sources/alerting/old-alerting/pause-an-alert-rule.md [hunk 3]
```diff
@@ -8,6 +8,8 @@ aliases = ["/docs/grafana/latest/alerting/pause-an-alert-rule/"]
 
 # Pause an alert rule
 
+>**Note**: Deprecated. Refer to [Grafana 8 alerting]({{< relref "../unified-alerting/_index.md" >}}) for documentation
```
Comment (achatterjee-grafana) [inline docs/sources/alerting/old-alerting/pause-an-alert-rule.md:11]: Not needed. A message on the landing page for old alerts is sufficient.

File: docs/sources/alerting/old-alerting/notifications.md [hunk 4]
```diff
@@ -8,6 +8,8 @@ aliases = ["/docs/grafana/latest/alerting/notifications/"]
 
 # Alert notifications
 
+>**Note**: Deprecated. Refer to [Grafana 8 alerting]({{< relref "../unified-alerting/_index.md" >}}) for documentation
```
Comment (achatterjee-grafana) [inline docs/sources/alerting/old-alerting/notifications.md:11]: Not sure what is deprecated here?

Comment (achatterjee-grafana) [inline docs/sources/alerting/old-alerting/notifications.md:11]: Not needed. A message on the landing page for old alerts is sufficient.

File: docs/sources/alerting/old-alerting/create-alerts.md [hunk 5]
```diff
@@ -8,6 +8,8 @@ aliases = ["/docs/grafana/latest/alerting/create-alerts/"]
 
 # Create alerts
 
+>**Note**: Deprecated. Refer to [Grafana 8 alerting]({{< relref "../unified-alerting/_index.md" >}}) for documentation
```
Comment (achatterjee-grafana) [inline docs/sources/alerting/old-alerting/create-alerts.md:11]: Not needed. A message on the landing page for old alerts is sufficient.

**Review by achatterjee-grafana — COMMENTED** [review]
Added review and requested a few clarifications.

File: docs/sources/alerting/unified-alerting/migration.md (hunk 1 above)
Comment (achatterjee-grafana) [reply docs/sources/alerting/unified-alerting/migration.md:7]: Answering my question. I spoke to @pkolyvas, and users with older OSS and Enterprise installations as well as users with Enterprise will still have legacy dashboard alerting OOTB. Let's keep this info and not remove it.

I am working on alerting docs in another PR and will make the necessary changes. Thanks!

**Comment by yuri-tceretian** [comment]
Closing in favor of https://github.com/grafana/grafana/pull/42262

**Pull request closed without merging**